src/app/child-dev-project/children/aser/model/aser.ts

```diff
@@ -58,40 +34,44 @@ export class Aser extends Entity {
     dataType: "configurable-enum",
     innerDataType: "reading-levels",
   })
-  hindi: ReadingLevel;
+  hindi: SkillLevel;
   @DatabaseField({
     label: $localize`:Label of the Bengali ASER result:Bengali`,
     dataType: "configurable-enum",
     innerDataType: "reading-levels",
   })
-  bengali: ReadingLevel;
+  bengali: SkillLevel;
   @DatabaseField({
     label: $localize`:Label of the English ASER result:English`,
     dataType: "configurable-enum",
     innerDataType: "reading-levels",
   })
-  english: ReadingLevel;
+  english: SkillLevel;
   @DatabaseField({
     label: $localize`:Label of the Math ASER result:Math`,
     dataType: "configurable-enum",
     innerDataType: "math-levels",
   })
-  math: MathLevel;
+  math: SkillLevel;
   @DatabaseField({
     label: $localize`:Label for the remarks of a ASER result:Remarks`,
   })
   remarks: string = "";
 
+  otherProp = "otherProp";
   getWarningLevel(): WarningLevel {
-    if (
-      Aser.isReadingPassedOrNA(this.english) &&
-      Aser.isReadingPassedOrNA(this.hindi) &&
-      Aser.isReadingPassedOrNA(this.bengali) &&
-      Aser.isMathPassedOrNA(this.math)
-    ) {
+    if (this.hasPassedEverything()) {
       return WarningLevel.OK;
     } else {
       return WarningLevel.WARNING;
     }
   }
+
+  private hasPassedEverything(): boolean {
+    const configurableEnum = new ConfigurableEnumDatatype(undefined).name;
+    const schema = this.getSchema();
+    return Object.keys(this)
+      .filter((key) => schema.get(key)?.dataType === configurableEnum)
+      .every((key) => !!(this[key] as SkillLevel).passed);
```

Comment:
the previous definition used to ignore subjects that were still empty (e.g. if for some kids "bengali" is not a relevant language). The new implementation seems the require all in an explicit pass state?